src/pages/Compare/index.spec.tsx

```diff
@@ -1361,4 +1380,133 @@ describe(__filename, () => {
       `Compare ${name}: ${baseVersionId}...${headVersionId}`,
     );
   });
+
+  describe('updateSelectedPath', () => {
+    const setUpForPathUpdateAndRender = ({
+      baseVersionId,
+      headVersionId,
+      initialPath,
+      lastSelectedPath,
+      location,
+      pathList,
+    }: {
+      baseVersionId: number;
+      headVersionId: number;
+      initialPath: string;
+      lastSelectedPath: string;
+      location: Location;
+      pathList: string[];
+    }) => {
+      const history = createFakeHistory({ location });
+      const version = createExternalVersionWithEntries(
+        pathList.map((path) => ({ path })),
+        { id: headVersionId, selected_file: initialPath },
+      );
+
+      return loadDiffAndRender({
+        baseVersionId,
+        headVersionId,
+        history,
+        lastSelectedPath,
+        version,
+      });
+    };
+
+    it('updates the selected path if it is different from the path in the URL', () => {
+      const baseVersionId = nextUniqueId();
+      const headVersionId = baseVersionId + 1;
+      const path1 = 'a/path';
+      const path2 = 'b/path';
+      const location = createFakeLocation({
+        search: queryString.stringify({ path: path2 }),
+      });
+
+      const { dispatchSpy } = setUpForPathUpdateAndRender({
+        baseVersionId,
+        headVersionId,
+        initialPath: path1,
+        lastSelectedPath: path1,
+        location,
+        pathList: [path1, path2],
+      });
+
+      expect(dispatchSpy).toHaveBeenCalledWith(
+        versionsActions.updateSelectedPath({
+          selectedPath: path2,
+          versionId: headVersionId,
+        }),
+      );
+    });
+
+    it('updates the selected path to the initial value if the path is undefined in the URL', () => {
+      const baseVersionId = nextUniqueId();
+      const headVersionId = baseVersionId + 1;
+      const initialPath = 'a/path';
+      const lastSelectedPath = 'b/path';
+
+      const { dispatchSpy } = setUpForPathUpdateAndRender({
+        baseVersionId,
+        headVersionId,
+        initialPath,
+        lastSelectedPath,
+        location: createFakeLocation({ search: '' }),
+        pathList: [initialPath, lastSelectedPath],
+      });
+
+      expect(dispatchSpy).toHaveBeenCalledWith(
+        versionsActions.updateSelectedPath({
+          selectedPath: initialPath,
+          versionId: headVersionId,
+        }),
+      );
+    });
+
+    it('does not update the selected path if it has been updated', () => {
```

Comment:
This test description doesn't seem clear to me. Did you mean to say "if it has *not* been updated"? Would it be better to say something like `does not update the selected path if it is the same as the path in the URL`?